packages/runtime-version/lib/RuntimeVersion.js

```diff
@@ -56,49 +75,153 @@ class RuntimeVersion {
+    } else if (options.lts) {
+      releaseType = ReleaseType.lts;
     }
-    return this.padVersionString_(version);
+
+    const host = options.ampUrlPrefix ? options.ampUrlPrefix.replace(/\/$/, '') : AMP_CACHE_HOST;
+
+    let rtv = await this.getVersionFromRuntimeMetadata_(host, releaseType);
+    if (!rtv && releaseType === ReleaseType.prod) {
+      rtv = await this.getVersionFromVersionTxt_(host, releaseType);
+    }
+
+    return rtv;
   }
 
   /* PRIVATE */
-  padVersionString_(version) {
-    return this.pad_(version, 15, 0);
+
+  /**
+   * Get runtime version from <host>/rtv/metadata
+   *
+   * @param {string} host - runtime host.
+   * @param {ReleaseType} releaseType - release type.
+   * @returns {Promise<string>} a promise containing the runtime version.
+   */
+  async getVersionFromRuntimeMetadata_(host, releaseType) {
+    const runtimeMetaUrl = host + RUNTIME_METADATA_PATH;
+    log.debug(`Fetching version from ${runtimeMetaUrl}`);
+
+    let response;
+    try {
+      response = await this.fetch_(runtimeMetaUrl);
+    } catch (ex) {}
+    if (!response || !response.ok) {
+      log.debug('RTV metadata endpoint could not be reached');
+      return;
+    }
+
+    let data;
+    try {
+      data = await response.json();
+    } catch (ex) {
+      log.debug('RTV metadata JSON malformed');
+      return;
+    }
+
+    let rtv;
+    if (releaseType === ReleaseType.canary) {
+      if (
+        Array.isArray(data.diversions) &&
+        data.diversions[0] &&
+        data.diversions[0].startsWith(this.getRtvConfigCode_(releaseType))
+      ) {
+        rtv = data.diversions[0];
+      }
+      if (!rtv) {
+        log.debug('RTV metadata JSON malformed, canary version not in diversions array');
+      }
+    } else if (releaseType === ReleaseType.lts) {
+      rtv = data.ltsRuntimeVersion;
+      if (!rtv) {
+        log.debug('RTV metadata JSON malformed, lts version not in ltsRuntimeVersion');
+      }
+    } else if (releaseType === ReleaseType.prod) {
+      rtv = data.ampRuntimeVersion;
+      if (!rtv) {
+        log.debug('RTV metadata JSON malformed, production version not in ampRuntimeVersion');
+      }
+    }
+
+    return rtv;
+  }
+
+  /**
+   * Get runtime version from <host>/version.txt, manually prepending
+   * production release code '01'. This method cannot be used to detect
+   * canary or lts releases.
+   *
+   * @param {string} host - runtime host.
+   * @param {ReleaseType} releaseType - release type.
+   * @returns {Promise<string>} a promise containing the runtime version.
+   */
+  async getVersionFromVersionTxt_(host, releaseType) {
+    if (releaseType !== ReleaseType.prod) {
+      log.debug(`version.txt lookup only supported for prod releases`);
+      return;
+    }
+
+    let versionTxtUrl = host + VERSION_TXT_PATH;
+    log.debug(`Falling back to ${versionTxtUrl}`);
+
+    let response;
+    try {
+      response = await this.fetch_(versionTxtUrl);
+    } catch (ex) {}
```

Comment:
please add a comment explaining why we can ignore the exception

Reply:
Comment added. Thanks for reviewing!